File: frontend/src/types/report.ts
export type WorkCategory =
  | 'Daily Work'
  | 'Customer Visit'
  | 'Trial'
  | 'Time Estimation'
  | 'Technical Support'
  | 'Service Support'
  | 'Internal Meeting'
  | 'Documentation'
  | 'Follow-up'
  | 'Other';

export type WorkStatus =
  | 'Draft'
  | 'In Progress'
  | 'Completed'
  | 'Pending'
  | 'Waiting for Customer'
  | 'Waiting for Internal Support';

export type WorkPriority = 'Low' | 'Medium' | 'High' | 'Critical';

export type WorkReport = {
  id: number;
  report_date: string;
  work_category: WorkCategory;
  customer_name: string | null;
  project_name: string | null;
  location: string | null;
  machine_model: string | null;
  activity_description: string;
  status: WorkStatus;
  priority: WorkPriority;
  pending_actions: string | null;
  next_follow_up_date: string | null;
  remarks: string | null;
  attachment_count: number;
  created_by: number;
  created_at: string;
  updated_at: string;
};

export type ReportInput = Omit<
  WorkReport,
  'id' | 'attachment_count' | 'created_by' | 'created_at' | 'updated_at'
>;

export type ReportListResponse = {
  items: WorkReport[];
  total: number;
};

export type DashboardSummary = {
  total_reports: number;
  reports_today: number;
  reports_this_week: number;
  pending_reports: number;
  recent_reports: WorkReport[];
};

export type Attachment = {
  id: number;
  report_id: number;
  file_name: string;
  file_type: string;
  file_path: string;
  file_size: number;
  uploaded_at: string;
};

File: backend/app/schemas/report.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field

WORK_CATEGORIES = [
    "Daily Work",
    "Customer Visit",
    "Trial",
    "Time Estimation",
    "Technical Support",
    "Service Support",
    "Internal Meeting",
    "Documentation",
    "Follow-up",
    "Other",
]

STATUSES = [
    "Draft",
    "In Progress",
    "Completed",
    "Pending",
    "Waiting for Customer",
    "Waiting for Internal Support",
]

PRIORITIES = ["Low", "Medium", "High", "Critical"]

StatusType = Literal[
    "Draft",
    "In Progress",
    "Completed",
    "Pending",
    "Waiting for Customer",
    "Waiting for Internal Support",
]
PriorityType = Literal["Low", "Medium", "High", "Critical"]


class WorkReportBase(BaseModel):
    report_date: date
    work_category: str = Field(pattern="^(" + "|".join(WORK_CATEGORIES) + ")$")
    customer_name: str | None = Field(default=None, max_length=120)
    project_name: str | None = Field(default=None, max_length=120)
    location: str | None = Field(default=None, max_length=120)
    machine_model: str | None = Field(default=None, max_length=120)
    activity_description: str = Field(min_length=3)
    status: StatusType
    priority: PriorityType
    pending_actions: str | None = None
    next_follow_up_date: date | None = None
    remarks: str | None = None


class WorkReportCreate(WorkReportBase):
    pass


class WorkReportUpdate(WorkReportBase):
    pass


class WorkReportResponse(WorkReportBase):
    id: int
    attachment_count: int
    created_by: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class WorkReportListResponse(BaseModel):
    items: list[WorkReportResponse]
    total: int
